code/modules/projectiles/guns/projectile/revolver.dm

```diff
@@ -363,26 +363,56 @@
 	else
 		return ..()
 
+
+
 /obj/item/gun/projectile/revolver/doublebarrel/sawoff(mob/user)
 	. = ..()
 	weapon_weight = WEAPON_MEDIUM
 
 /obj/item/gun/projectile/revolver/doublebarrel/attack_self(mob/living/user)
 	var/num_unloaded = 0
+	var/obj/item/storage/belt/bandolier/B
+	for(var/obj/item/I in user.contents)
+		if(istype(I, /obj/item/storage/belt/bandolier))
+			B = I
+			break
```

Comment:
I think it may be better to just check their belt slot, if you do want to iterate through stuff, maybe just have it be a list of their belt slot and hands slots